clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/BaseOperation.java

```diff
@@ -16,26 +16,140 @@
 
 package org.ehcache.clustered.client.internal.store.operations;
 
+import org.ehcache.clustered.client.internal.store.operations.codecs.CodecException;
+import org.ehcache.spi.serialization.Serializer;
+
+import java.nio.ByteBuffer;
+
 /**
  * Base class that represents a {@link org.ehcache.Cache} operation
  *
  * @param <K> key type
  */
-public abstract class BaseOperation<K> implements Operation<K> {
+public abstract class BaseOperation<K, V> implements Operation<K, V> {
+
+  public static final int BYTE_SIZE_BYTES = 1;
+  public static final int INT_SIZE_BYTES = 4;
+  public static final int LONG_SIZE_BYTES = 8;
 
-  protected final K key;
+  protected K key;
+  protected V value;
+
+  public BaseOperation() {
+    this.key = null;
+    this.value = null;
+  }
 
   public BaseOperation(final K key) {
+    this(key, null);
```

Comment:
This constructor makes no sense, as it will always throw the NPE because the value is null.